### PR TITLE
Remove PrivateRegistry from test suite and some e2e tests that depend on it

### PR DESCRIPTION
**Pull request by dims**
Related to https://github.com/kubernetes/kubernetes/issues/97026

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
manifest of list of registries for e2e test suite no longer needs a private registry
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
lgtm from sig-apps pov

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -227,7 +223,6 @@ const (
 func initImageConfigs(list RegistryList) (map[ImageID]Config, map[ImageID]Config) {
 	configs := map[ImageID]Config{}
 	configs[Agnhost] = Config{list.PromoterE2eRegistry, "agnhost", "2.52"}
-	configs[AgnhostPrivate] = Config{list.PrivateRegistry, "agnhost", "2.6"}
```
[the remark this answers is not in the document]
Comment (dims) [reply test/utils/image/manifest.go:230]: How to Play Fruit Ninja Gameplay

File: test/e2e/common/node/image_credential_provider.go [hunk 2]
```diff
@@ -1,69 +0,0 @@
-/*
-Copyright 2022 The Kubernetes Authors.
```
Comment (SergeyKanzhelev) [inline test/e2e/common/node/image_credential_provider.go:2]: @ndixita is there any other tests covering creds provider e2e? 

/hold

Comment (ndixita) [reply test/e2e/common/node/image_credential_provider.go:2]: there are no other tests covering the credential provider.

**Pull request closed without merging**